py/server/deephaven/barrage.py

```diff
@@ -0,0 +1,187 @@
+                    port: int = 10000,
+                    auth_type: str = "Anonymous",
+                    auth_token: str = "",
+                    use_tls: bool = False,
+                    tls_root_certs: bytes = None,
+                    ) -> BarrageSession:
+    """Returns a Deephaven gRPC session to a remote server if a cached session is available; otherwise, creates a new
+    session.
+
+    Note: client authentication is not supported yet.
+
+    Args:
+        host (str): the host name or IP address of the Deephaven server.
+        port (int): the port number that the remote Deephaven server is listening on, default is 10000.
+        auth_type (str): the authentication type string, can be "Anonymous', 'Basic", or any custom-built
+            authenticator in the server, such as "io.deephaven.authentication.psk.PskAuthenticationHandler",
+            default is 'Anonymous'.
+        auth_token (str): the authentication token string. When auth_type is 'Basic', it must be
+            "user:password"; when auth_type is "Anonymous', it will be ignored; when auth_type is a custom-built
+            authenticator, it must conform to the specific requirement of the authenticator
+        use_tls (bool): if True, use a TLS connection.  Defaults to False
+        tls_root_certs (bytes): PEM encoded root certificates to use for TLS connection, or None to use system defaults.
+             If not None implies use a TLS connection and the use_tls argument should have been passed
+             as True. Defaults to None
+
+    Returns:
+        a Deephaven Barrage session
+
+    Raises:
+        DHError
+    """
+    try:
+        if tls_root_certs and not use_tls:
+            raise DHError(message="tls_root_certs is provided but use_tls is False")
+
+        target_uri = f"{host}:{port}"
+        if use_tls:
+            target_uri = f"dh://{target_uri}"
+        else:
+            target_uri = f"dh+plain://{target_uri}"
+
+        j_client_config = _build_client_config(target_uri, tls_root_certs)
+        auth = f"{auth_type} {auth_token}"
+
+        try:
+            return _get_barrage_session_via_api_server(j_client_config, auth)
+        except:
+            # fall back to the direct way when we don't have a fully initialized server, used for testing only
+            # TODO: remove when we are done with restructuring the integrations tests wiring https://github.com/deephaven/deephaven-core/issues/5401
+            return _get_barrage_session_direct(j_client_config, auth)
+    except Exception as e:
+        raise DHError(e, "failed to get a barrage session to the target remote Deephaven server.") from e
+
+
+def _get_barrage_session_via_api_server(client_config: jpy.JType, auth: str) -> BarrageSession:
+    j_barrage_session_factory_creator = _JDeephavenApiServer.getInstance().sessionFactoryCreator()
+    j_barrage_session_factory = j_barrage_session_factory_creator.barrageFactory(client_config, auth)
```

Comment:
This method does not exist; it is `public BarrageSessionFactoryConfig.Factory barrageFactory(ClientConfig clientConfig)`. After creating the factory, you'll need to call one of:

```
io.deephaven.client.impl.BarrageSessionFactoryConfig.Factory#newBarrageSession()
io.deephaven.client.impl.BarrageSessionFactoryConfig.Factory#newBarrageSession(io.deephaven.client.impl.SessionConfig)
```

The former is recommended if auth is not set. That also reminds me, we should _probably_ prefer 

```python
                    auth_type: str = None,
                    auth_token: str = None,
```

instead? Also, this is a different API than the server presents (where auth is represented with a single string).